packages/experimental/navigation/README.md

```diff
@@ -0,0 +1,4 @@
+## OTel Instrumentation Library - React Native / Navigation
```

Comment:
could add a few usage examples as well similar to something like https://github.com/open-telemetry/opentelemetry-js/tree/main/experimental/packages/opentelemetry-instrumentation-http to show how the different instrumentations would be used in an app

given we're introducing the "experimental" folder for the first time I think a quick `packages/experimental/README.md` would be worthwhile as well, we can chat on slack about what should go in there